core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -2178,23 +2180,36 @@ class WorkspaceService(protected val userInfo: UserInfo,
     )
   }
 
-  private def syncPolicies(workspaceId: String, policyEmailsByName: Map[SamResourcePolicyName, WorkbenchEmail], workspaceRequest: WorkspaceRequest, parentSpan: Span) = {
-    traceWithParent("traversePolicies", parentSpan) (s1 =>
-      Future.traverse(policyEmailsByName.keys) { policyName =>
-        if (policyName == SamWorkspacePolicyNames.projectOwner && workspaceRequest.authorizationDomain.getOrElse(Set.empty).isEmpty) {
-          // when there isn't an auth domain, we will use the billing project admin policy email directly on workspace
-          // resources instead of synching an extra group. This helps to keep the number of google groups a user is in below
-          // the limit of 2000
-          Future.successful(())
-        } else if (WorkspaceAccessLevels.withPolicyName(policyName.value).isDefined || policyName == SamWorkspacePolicyNames.canCompute) {
-          // only sync policies that have corresponding WorkspaceAccessLevels to google because only those are
-          // granted bucket access (and thus need a google group)
-          traceWithParent(s"syncPolicy-${policyName}", s1)(_ => samDAO.syncPolicyToGoogle(SamResourceTypeNames.workspace, workspaceId, policyName))
-        } else {
-          Future.successful(())
-        }
+  private def syncPolicies(workspaceId: String, policyEmailsByName: Map[SamResourcePolicyName, WorkbenchEmail], workspaceRequest: WorkspaceRequest, parentSpan: Span): Future[List[Any]] = {
+    // when there isn't an auth domain, we will use the billing project admin policy email directly on workspace
+    // resources instead of synching an extra group. This helps to keep the number of google groups a user is in below
+    // the limit of 2000
+    def hasAuthorizationDomain(request: WorkspaceRequest): Boolean =
+      request.authorizationDomain match {
+        case Some(xs) => !xs.isEmpty
+        case None => false
+      }
+
+    def isProjectOwner(policy: SamResourcePolicyName): Boolean =
+      policy == SamWorkspacePolicyNames.projectOwner
+
+    // only sync policies that have corresponding WorkspaceAccessLevels to google because only those are
+    // granted bucket access (and thus need a google group)
+    def isCanCompute(policy: SamResourcePolicyName): Boolean =
+      policy == SamWorkspacePolicyNames.canCompute
+
+    traceWithParent("traversePolicies", parentSpan) { s1 =>
+      // This used to be Future.traverse but that runs all the futures eagerly and concurrently. This led to a perf
+      // issue with hitting the akka subscription timeout. We switched to use cats traverse to make these calls serially.
+      policyEmailsByName.keys.toList.traverse { policyName =>
+        Applicative[Future].whenA(
```

Comment:
I thought you want this sync call to happen sequentially? in that case, you do want to convert these `Future` into `IO`, and run `traverse` (or `traverse_` if you're returning `IO[Unit]`) on`IO` instead of `Future`